src/Castle.Core/DynamicProxy/Generators/ClassProxyGenerator.cs

```diff
@@ -74,7 +74,9 @@ protected virtual Type GenerateType(string name, Type[] interfaces, INamingScope
 				// TODO: redo it
 				if (contributor is MixinContributor)
 				{
-					constructorArguments.AddRange((contributor as MixinContributor).Fields);
+                    // constructor arguments are ordered by interface name as the MixinsData does
+                    var orderedFields = (contributor as MixinContributor).Fields.OrderBy(x => x.Fieldbuilder.FieldType.Name);
```

Comment:
I'd much prefer `MixinContributor` to do the sorting rather than a proxy generator, what do you think about changing the `fields` field to a `SortedDictionary<Type, FieldReference>`?